src/memory/paging/entry.rs

```diff
@@ -12,6 +12,7 @@ use multiboot2::ElfSection;
 use memory::Frame;
 
 /// A representation of a page table entry.
+#[derive(Clone,PartialEq,Debug)]
```

Comment:
How will we make sure this doesn't result in a double free?

Reply:
I don't know how it would cause a double free... can you explain?

Reply:
An entry owns a physical frame, and when it's deallocated so is it's backing frame. If `Entry`s are clone, then a frame that is in use could be cloned and freed to the allocator, leading to memory corruption. There was a hack around this with `Frame`s in phil's blog.